File: 11st/pro_1.cpp
#include <string>
#include <vector>
#include <string>

using namespace std;

int a[3] = { 1,2,4 };
vector<int> v;

string solution(int n) {
    string answer = "";
    int leave;
    while (true) {
        n--;
        if (n < 3) {//³¡
            v.push_back(a[n % 3]);
            break;
        }
        v.push_back(a[n % 3]);
        n = n / 3;
    }

    for (int i = v.size() - 1; i >= 0; i--) {
        answer += to_string(v[i]);
    }
    return answer;
}

File: sort/pro_1.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String num[] = new String[numbers.length];
        int j=0;
        for(int i : numbers){
            num[j]=Integer.toString(i);
            j++;
        }
        Arrays.sort(num, new Comparator<String>(){
            @Override
            public int compare(String s1, String s2){
                return (s2+s1).compareTo(s1+s2);
            }
        });//Arrays.sort(num,(s1,s2)->(s2+s1).compareTo(s1+s2));도 가능
        if(num[0].equals("0"))  return "0";
        for(String i : num){
            answer+=i;
        }
        return answer;
    }
}